django/db/backends/oracle/base.py

```diff
@@ -528,14 +535,22 @@ def _fix_for_params(self, query, params, unify_by_values=False):
             # values. It can be used only in single query execute() because
             # executemany() shares the formatted query with each of the params
             # list. e.g. for input params = [0.75, 2, 0.75, 'sth', 0.75]
-            # params_dict = {0.75: ':arg0', 2: ':arg1', 'sth': ':arg2'}
+            # params_dict = {
+            #     (float, 0.75): ':arg0',
+            #     (int, 2): ':arg1',
+            #     (str, 'sth'): ':arg2',
+            # }
             # args = [':arg0', ':arg1', ':arg0', ':arg2', ':arg0']
             # params = {':arg0': 0.75, ':arg1': 2, ':arg2': 'sth'}
+            param_types = [(type(param), param) for param in params]
```

Comment:
We can no longer mix `boolean` and `number` in Oracle 23c, as it will raise:
```
django.db.utils.DatabaseError:
ORA-00932: expression is of data type BOOLEAN, which is incompatible with expected data type NUMBER
```
for all queries that use any combination of `0`, `1`, `True`, and `False` as parameters.